src/igvfd/tests/test_types_analysis_set.py

```diff
@@ -18,3 +18,82 @@ def test_calculated_donors(testapp, analysis_set_base, primary_cell, human_donor
+
+
+def test_assay_titles(testapp, analysis_set_base, measurement_set_mpra, measurement_set_multiome):
+    testapp.patch_json(
+        analysis_set_base['@id'],
+        {
+            'input_file_sets': [measurement_set_mpra['@id']]
+        }
+    )
+    res = testapp.get(analysis_set_base['@id'])
+    assert set(res.json.get('assay_titles')) == {'massively parallel reporter assay'}
+    testapp.patch_json(
+        measurement_set_mpra['@id'],
+        {
+            'preferred_assay_title': 'lentiMPRA'
+        }
+    )
+    res = testapp.get(analysis_set_base['@id'])
+    assert set(res.json.get('assay_titles')) == {'lentiMPRA'}
+    testapp.patch_json(
+        analysis_set_base['@id'],
+        {
+            'input_file_sets': [measurement_set_mpra['@id'],
+                                measurement_set_multiome['@id']]
+        }
+    )
+    res = testapp.get(analysis_set_base['@id'])
+    assert set(res.json.get('assay_titles')) == {'ATAC-seq', 'lentiMPRA'}
+
+
+def test_analysis_set_summary(testapp, analysis_set_base, base_auxiliary_set, measurement_set_mpra, measurement_set_multiome, primary_analysis_set):
+    # With no input_file_sets present, summary is based on analysis file_set_type only
+    res = testapp.get(analysis_set_base['@id']).json
+    assert res.get('summary', '') == 'intermediate analysis of data'
+    # When no MeasurementSets (even nested in AnalysisSets) are present, data for other FileSet types are included in the summary
+    testapp.patch_json(
+        analysis_set_base['@id'],
+        {
+            'input_file_sets': [base_auxiliary_set['@id']]
+        }
+    )
+    res = testapp.get(analysis_set_base['@id']).json
+    assert res.get('summary', '') == 'intermediate analysis of gRNA sequencing data'
+    testapp.patch_json(
+        analysis_set_base['@id'],
+        {
+            'input_file_sets': [base_auxiliary_set['@id'],
+                                measurement_set_mpra['@id'],
+                                measurement_set_multiome['@id']]
+        }
+    )
+    res = testapp.get(analysis_set_base['@id']).json
+    assert res.get('summary', '') == 'intermediate analysis of ATAC-seq, massively parallel reporter assay data'
+    # Preferred_assay_title of MeasurementSet is used instead of assay_term in summary whenever present
+    testapp.patch_json(
+        measurement_set_mpra['@id'],
+        {
+            'preferred_assay_title': 'lentiMPRA'
+        }
+    )
+    testapp.patch_json(
+        analysis_set_base['@id'],
+        {
+            'input_file_sets': [measurement_set_mpra['@id'],
+                                measurement_set_multiome['@id'],
+                                primary_analysis_set['@id']]
+        }
+    )
+    res = testapp.get(analysis_set_base['@id']).json
+    assert res.get('summary', '') == 'intermediate analysis of ATAC-seq, STARR-seq, lentiMPRA data'
+    # An infinite loop should be avoided
+    testapp.patch_json(
```

Comment:
This patch causes error: r`equest = <Request at 0x7f9634ecda50 PATCH http://localhost/analysis-sets/IGVFDS0682MRHA/> exc = RecursionError('maximum recursion depth exceeded')`. What kind of file set are you trying to generate here?

Reply:
I know that the patch is causing it. that is why I am saying it is not the summary calculation to blame. I am trying to create circular relationship .
Analysis A input_file_sets = [B]
Analysis B input_file_Sets = [A]